drivers/sdhc/sdhc_stm32.c

```diff
@@ -0,0 +1,659 @@
+/*
+ * Copyright (c) 2025 EXALT Technologies.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_sdhc
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+
+LOG_MODULE_REGISTER(sdhc_stm32, CONFIG_SDHC_LOG_LEVEL);
+
+#define SD_TIMEOUT         ((uint32_t)0x00100000U) /* Timeout value for SD card operations*/
+#define STM32_SDIO_F_MIN   SDMMC_CLOCK_400KHZ
+#define STM32_SDIO_F_MAX   (MHZ(208))  /* Maximum frequency supported by the SDIO interface */
+#define SDIO_OCR_SDIO_S18R (1U << 24U) /* SDIO OCR bit indicating support for 1.8V switching */
+
+struct sdhc_stm32_config {
+	bool hw_flow_control;              /* flag for enabling hardware flow control */
+	uint8_t bus_width;                 /* Width of the SDIO bus (1-bit or 4-bit mode) */
+	uint16_t clk_div;                  /* Clock divider value to configure SDIO clock speed */
+	uint32_t power_delay_ms;           /* power delay prop for the host */
+	const struct stm32_pclken *pclken; /* Pointer to peripheral clock configuration */
+	const struct pinctrl_dev_config *pcfg; /* Pointer to pin control configuration */
+	struct gpio_dt_spec sdhi_on_gpio; /* Power pin to control the regulators used by card.*/
+};
+
+struct sdhc_stm32_data {
+	struct k_mutex bus_mutex;     /* Sync between commands */
+	SDIO_HandleTypeDef hsd;       /* Pointer to I3C module base addr */
```

Comment:
Pointer to SDHC instance is rather static. Any reason to store it in data struct vs config ?